src/components/Notifications/NotificationSelect.tsx

```diff
@@ -0,0 +1,166 @@
+  onClickAddRegexSelector: () => void;
+  onChangeSelectedOption?: (optionValue: string) => void;
+  selectedOption?: string;
+}
+
+export const NotificationSelect: React.VFC<NotificationSelectProps> = ({
+  triggers,
+  extraFields,
+  selectedTriggerIndex,
+  showAddCriteria,
+  setSelectedTriggerIndex,
+  extraFieldInputVals,
+  setExtraFieldInputVals,
+  regexSelectorProps,
+  disableAddCriteria,
+  onClickAddRegexSelector,
+  onChangeSelectedOption,
+  selectedOption,
+}) => (
+  <Section>
+    <Body weight="medium">Choose an event</Body>
+    <SectionLabelContainer>
+      <InputLabel>Event</InputLabel>
+    </SectionLabelContainer>
+    <StyledSelect
+      value={selectedTriggerIndex}
+      onChange={(v: number) => {
+        setSelectedTriggerIndex(v);
+      }}
+      data-cy="event-select"
+    >
+      {triggers.map((t, i) => (
+        <Option
+          key={`trigger_${t.trigger}_${t.resourceType}_${t.payloadResourceIdKey}`}
+          value={i}
+          data-cy={`trigger_${i}-option`}
+        >
+          {t.label}
+        </Option>
+      ))}
+    </StyledSelect>
+    {extraFields &&
+      extraFields.map(({ text, key, type, dataCy: inputDataCy, options }) => (
+        <ExtraFieldContainer key={key}>
+          <SectionLabelContainer>
+            <InputLabel htmlFor={`${key}-input`}>{text}</InputLabel>
+          </SectionLabelContainer>
+
+          {type === "select" ? (
+            <StyledSelect
+              value={selectedOption}
+              onChange={onChangeSelectedOption}
+              data-cy="when-select"
+            >
+              {options.map((o) => (
+                <Option value={o} key={o}>
+                  {o}
+                </Option>
+              ))}
+            </StyledSelect>
+          ) : (
+            <StyledInput
+              data-cy={inputDataCy}
+              id={`${key}-input`}
+              onChange={(event) => {
+                setExtraFieldInputVals({
+                  ...extraFieldInputVals,
+                  [key]: event.target.value,
+                });
+              }}
+              value={extraFieldInputVals[key]}
+            />
+          )}
+        </ExtraFieldContainer>
+      ))}
+    {showAddCriteria && (
+      <>
+        <RegexSelectorInputContainer>
+          {regexSelectorProps.map((props, i) => (
+            <RegexSelectorInput
+              canDelete
+              // eslint-disable-next-line react/no-array-index-key
+              key={`${props.dataCyPrefix}_${i}`}
+              dataCyPrefix={i}
+              {...props}
+            />
+          ))}
+        </RegexSelectorInputContainer>
+        <Button
+          data-cy="add-regex-selector-button"
+          disabled={disableAddCriteria}
+          onClick={onClickAddRegexSelector}
+        >
+          <Icon glyph="Plus" />
+          Add additional criteria
+        </Button>
+      </>
+    )}
+  </Section>
+);
+
+export const inputWidth = "width: calc(80% - 55px);";
```

Comment:
I'm not a huge fan of putting shared styles into a component file. Could we move these to a new file named `style.ts` in the same folder?

Reply:
I'll keep it open to do in the next PR, since I'll be adding more styles I'm sure anyway